fs_attachment/models/ir_attachment.py

```diff
@@ -381,6 +383,13 @@ def _storage_file_read(self, fname: str) -> bytes | None:
             )
         return b""
 
+    def _storage_write_option(self, fs):
+        if hasattr(fs.fs, "s3"):
+            # For S3 storage the mimetype should be pass in the kwargs
+            return {"ContentType": self._context["mimetype"]}
```

Comment:
the s3 fs could be nested at a deeper level... IMO you should recursively test the nested fs. Something like:

```
_fs = fs
while _fs:
    if hasattr(_fs, "s3"):
        return {"ContentType": self._context["mimetype"]}
    _fs = getattr(_fs, "fs", None)
```